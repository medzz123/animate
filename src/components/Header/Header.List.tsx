import { LinkHighlight } from '@components/Highlight';
import Link from 'next/link';
import React, { FunctionComponent } from 'react';

import { LinkListProps } from './Header.models';
import { List, ThemeButton } from './Header.styles';

const navigationLinks = [
  { path: '/browse', title: 'browse', id: 'browse-id-1' },
  { path: '/animate', title: 'animate', id: 'animate-id-1' },
];

const LinkList: FunctionComponent<LinkListProps> = ({
  toggleDarkMode,
  vertical,
}) => {
  return (
    <List vertical={vertical}>
      {navigationLinks.map((item) => (
        <li key={item.id}>
          <Link href={item.path}>
            <a>
              <LinkHighlight>{item.title}</LinkHighlight>
            </a>
          </Link>
        </li>
      ))}
      <ThemeButton type="button" onClick={toggleDarkMode}>
        <LinkHighlight>toggle theme</LinkHighlight>
      </ThemeButton>
    </List>
  );
};

export default LinkList;
