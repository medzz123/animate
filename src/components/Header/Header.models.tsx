export interface HeaderProps {
  /**
   * Toggle theme
   **/
  toggleDarkMode: () => void;
}

export interface LinkListProps {
  vertical: boolean;
  toggleDarkMode: () => void;
}

export interface StyleProps {
  vertical?: boolean;
}
