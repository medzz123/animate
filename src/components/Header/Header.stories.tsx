// import { Meta, Story } from '@storybook/react/types-6-0';
// import React from 'react';

// import Header from './Header';

// export default {
//   title: 'Components/Header',
//   component: Header,
// } as Meta;

// const Template: Story = (storyArguments) => {
//   return <Header {...storyArguments} />;
// };

// export const HeaderStory = Template.bind({});
