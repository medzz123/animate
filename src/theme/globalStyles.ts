import { createGlobalStyle, css } from 'styled-components';

const styles = css`
  html {
    font-family: 'Mulish', sans-serif;
    scroll-behavior: smooth;
  }

  body,
  html {
    margin: 0;
    padding: 0;
    background-color: ${(p) => p.theme.white};
  }

  * {
    box-sizing: border-box;
  }

  ul {
    list-style: none;
  }

  body {
    font-family: 'Mulish', sans-serif;
    color: ${(p) => p.theme.black};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: 'Abril Fatface', sans-serif;
    color: ${(p) => p.theme.black};
    margin: 0;
  }

  h1 {
    font-size: 5vh;
    font-weight: 600;
  }

  @media screen and (max-width: 996px) {
    h1 {
      font-size: 50px;
    }
  }

  @media screen and (max-width: 479px) {
    h1 {
      font-size: 37px;
    }
  }

  a {
    text-decoration: none;
    color: inherit;
  }

  p {
    font-size: 18px;
    line-height: 32px;
    margin: 0;
  }

  p,
  span {
    color: ${(p) => p.theme.black};
  }
`;

export const GlobalStyle = createGlobalStyle`
  ${styles}
`;
