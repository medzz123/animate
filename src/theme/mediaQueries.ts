import { tokens } from './tokens';
/**
 * Media query function which will return a min-width media query with the breakpoint provided
 */
export const mq = (level: keyof typeof tokens.breakpoints) =>
  `@media (min-width: ${tokens.breakpoints[level]})`;
