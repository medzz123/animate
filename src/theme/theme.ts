export interface Theme {
  black: string;
  white: string;
  accent: string;
  background: string;
}

const light: Theme = {
  black: '#000000',
  white: '#ffffff',
  accent: '#FF3366',
  background: '#e9ecef',
};

const dark: Theme = {
  black: '#ffffff',
  white: '#000000',
  accent: '#FF3366',
  background: '#263238',
};

const green: Theme = {
  black: '#ffffff',
  white: '#000000',
  accent: '#1de9b6',
  background: '#263238',
};

const blue: Theme = {
  black: '#ffffff',
  white: '#000000',
  accent: '#2979ff',
  background: '#263238',
};

export const themes = {
  light,
  dark,
  green,
  blue,
};
