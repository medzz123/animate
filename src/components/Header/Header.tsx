import DashboardIcon from '@material-ui/icons/Dashboard';
import Link from 'next/link';
import React, { FunctionComponent } from 'react';

import { HeaderContainer, LogoLink, MainLink } from './Header.styles';

const Header: FunctionComponent = () => {
  return (
    <HeaderContainer data-testid="header">
      <Link href="/">
        <LogoLink tabIndex={0}>Animate.</LogoLink>
      </Link>

      <ul>
        <li>
          <Link href="/browse">
            <a>Browse</a>
          </Link>
        </li>
        <li>
          <Link href="/animate">
            <MainLink tabIndex={0}>
              <DashboardIcon /> Animate
            </MainLink>
          </Link>
        </li>
      </ul>
    </HeaderContainer>
  );
};

export default Header;
