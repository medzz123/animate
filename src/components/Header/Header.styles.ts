import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { tokens } from '../../theme/tokens';

export const HeaderContainer = styled.div`
  padding: ${tokens.sizes[12]} ${tokens.sizes[12]};
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const LogoLink = styled(Link)`
  font-weight: 900;
  font-size: ${tokens.sizes[22]};
  cursor: pointer;
  transition: color 200ms ease;
  color: ${(p) => p.theme.paragraph};
  margin-left: 0;
`;

export const MainLink = styled(Link)`
  color: ${(p) => p.theme.paragraph};

  margin-right: 0;
`;
