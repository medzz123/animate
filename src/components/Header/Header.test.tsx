import { theme } from '@theme/theme';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Header from './Header';

describe('<Header />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
