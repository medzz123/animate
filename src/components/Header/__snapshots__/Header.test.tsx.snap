// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test 1`] = `
<div>
  <div
    class="sc-bdvvaa blkmKL"
    data-testid="header"
  >
    <a
      class="sc-gsDJrp juuEKj"
      href="/"
      tabindex="0"
    >
      Animate.
    </a>
    <ul>
      <li>
        <a
          href="/browse"
        >
          Browse
        </a>
      </li>
      <li>
        <a
          class="sc-dkPtyc eTdaSJ"
          href="/animate"
          tabindex="0"
        >
          Editor
        </a>
      </li>
    </ul>
  </div>
</div>
`;
